core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -619,11 +620,16 @@ protected Type visitDereferenceExpression(DereferenceExpression node, StackableA
             RowType rowType = (RowType) baseType;
             String fieldName = node.getField().getValue();
 
+            int index = -1;
             Type rowFieldType = null;
             for (RowType.Field rowField : rowType.getFields()) {
-                if (fieldName.equalsIgnoreCase(rowField.getName().orElse(null))) {
+                String rowFieldName = rowField.getName().orElse(null);
+                if (fieldName.equalsIgnoreCase(rowFieldName)) {
+                    if (index >= 0) {
+                        throw ambiguousAttributeException(node, QualifiedName.of(rowFieldName));
```

Comment:
I would do:
```
throw semanticException(AMBIGUOUS_NAME, node.getField(), "Ambiguous row field reference: " + fieldName);
```
to adjust the error message, and also to locate the error at the ambiguous identifier.